app/containers/Admin/ApplicationReview/ReviewSidebar.tsx

```diff
@@ -109,14 +149,48 @@ export const ReviewSidebar: React.FunctionComponent<Props> = ({
         description={description}
         createdAt={createdAt}
         createdBy={`${ciipUserByCreatedBy.firstName} ${ciipUserByCreatedBy.lastName}`}
-        viewOnly={applicationReviewStep.isComplete}
+        viewOnly={isFinalized || applicationReviewStep?.isComplete}
         isResolved={resolved}
         onResolveToggle={resolveComment}
         onDelete={(id) => deleteComment(id, commentType)}
       />
     );
   };
 
+  const confirmCommentsResolvedModal = (
+    <GenericConfirmationModal
+      show={showUnresolvedCommentsModal}
+      title={`Confirm ${applicationReviewStep?.reviewStepByReviewStepId?.stepName} review is completed`}
+      onConfirm={() => {
+        markReviewStepComplete(true);
+        setShowUnresolvedCommentsModal(false);
+      }}
+      size="lg"
+      onCancel={() => setShowUnresolvedCommentsModal(false)}
+      confirmButtonVariant="outline-primary"
+      cancelButtonVariant="primary"
+      cancelButtonText="Cancel and resolve comments"
```

Comment:
Clicking on this button only cancels, it doesn't resolve comments. Should the text just be `Cancel`?

Reply:
That button is intended to do nothing - analysts should manually resolve individual comments to ensure they are indeed Resolved. The text is meant to instruct the user what to do next (1. Cancel 2. Make sure your comments are resolved).

Reply:
It's supposed to mean "I want to cancel, I will resolve comments before proceeding" but that was too long 😛

Reply:
I did have a second where I was wondering if it was going to automatically resolve the comments when I click the button.
 If you think it's clear as-is then I'm happy to leave it like that.

Reply:
I could see how one might take it the other way though - perhaps just Cancel is safer.

**Side note**: It is still possible to proceed and mark a step completed without resolving all comments - this allows them to write comments about why an application was Rejected for example. It's just a checkpoint they requested to remind them to ensure they've addressed any unresolved comments.

Reply:
re: Side note, the modal text was very clear about that :+1: